src/components/proxy/packet_router.rs

```diff
@@ -73,6 +73,7 @@ impl<P: PacketMut> DownstreamPacket<P> {
         config: &Arc<Config>,
         sessions: &S,
         destinations: &mut Vec<crate::net::EndpointAddress>,
+        deny_new_sessions: bool,
```

Comment:
Hmm, I'm not sure if we should disallow new sessions as that could cause disruption for players who have an IP:port change in the middle of a session while a proxy is being shut down. I think a better alternative would be to have a "termination timeout" that can specified in the CLI, where if enabled it would shutdown now matter what after a specified number of seconds. If not specified it should have no timeout, because for example in k8s we already specify a graceful termination time of 40 minutes to give time for games to finish before it has to shutdown.

Reply:
Ok.